service/frontend/wrappers/clusterredirection/policy.go

```diff
@@ -438,44 +438,27 @@ func (policy *selectedOrAllAPIsForwardingRedirectionPolicy) activeClusterForActi
 
-	if workflowExecution == nil {
-		policy.logger.Debug("Workflow execution is nil, using active cluster in the same region", tag.WorkflowDomainName(domainEntry.GetInfo().Name), tag.OperationName(apiName))
-		return policy.activeClusterInSameRegion(ctx, domainEntry, policy.currentClusterName)
+	if workflowExecution == nil || workflowExecution.WorkflowID == "" {
+		policy.logger.Debug("Workflow execution is nil, using current cluster", tag.WorkflowDomainName(domainEntry.GetInfo().Name), tag.OperationName(apiName))
+		return policy.currentClusterName
 	}
 	if workflowExecution.RunID == "" {
-		policy.logger.Debug("Workflow execution run id is empty, using active cluster in the same region", tag.WorkflowDomainName(domainEntry.GetInfo().Name), tag.OperationName(apiName))
-		return policy.activeClusterInSameRegion(ctx, domainEntry, policy.currentClusterName)
+		policy.logger.Debug("Workflow execution run id is empty, using current cluster", tag.WorkflowDomainName(domainEntry.GetInfo().Name), tag.OperationName(apiName))
+		return policy.currentClusterName
 	}
 
 	lookupRes, err := policy.activeClusterManager.LookupWorkflow(ctx, domainEntry.GetInfo().ID, workflowExecution.WorkflowID, workflowExecution.RunID)
 	if err != nil {
-		policy.logger.Error("Failed to lookup active cluster of workflow, using active cluster in the same region", tag.WorkflowDomainName(domainEntry.GetInfo().Name), tag.WorkflowID(workflowExecution.WorkflowID), tag.WorkflowRunID(workflowExecution.RunID), tag.OperationName(apiName), tag.Error(err))
-		return policy.activeClusterInSameRegion(ctx, domainEntry, policy.currentClusterName)
+		policy.logger.Error("Failed to lookup active cluster of workflow, using current cluster", tag.WorkflowDomainName(domainEntry.GetInfo().Name), tag.WorkflowID(workflowExecution.WorkflowID), tag.WorkflowRunID(workflowExecution.RunID), tag.OperationName(apiName), tag.Error(err))
+		return policy.currentClusterName
 	}
 
 	policy.logger.Debug("Lookup workflow result for active-active domain request", tag.WorkflowDomainName(domainEntry.GetInfo().Name), tag.WorkflowID(workflowExecution.WorkflowID), tag.WorkflowRunID(workflowExecution.RunID), tag.OperationName(apiName), tag.ActiveClusterName(lookupRes.ClusterName))
 	return lookupRes.ClusterName
 }
-
-// activeClusterInSameRegion returns the active cluster in the same region as the given cluster name
-// If the lookup fails, it returns current cluster name as fallback
-// In that case, if current cluster is not the active cluster the caller will return an not-active error
-// TODO(active-active): Revisit this logic for strong-consistency requests and potentially reject them instead of serving them from the current cluster
-func (policy *selectedOrAllAPIsForwardingRedirectionPolicy) activeClusterInSameRegion(
-	ctx context.Context,
-	domainEntry *cache.DomainCacheEntry,
-	clusterName string,
-) string {
-	lookupRes, err := policy.activeClusterManager.LookupCluster(ctx, domainEntry.GetInfo().ID, clusterName)
```

Comment:
you can also remove `LookupCluster` from manager interface/impl

Reply:
Yeah, I'll do it in a separate PR. I also want to do some refactoring when implementing/fixing lookups for external entity.